ObjectOrientedProgramming/Memory Ownership/Ownership/task.md

```diff
@@ -0,0 +1,49 @@
+Memory Ownership in C++
+In C++, memory ownership refers to the responsibility of managing the allocation and deallocation of memory for objects. Understanding memory ownership is crucial to prevent memory leaks, avoid double-deletions, and ensure proper resource management. The ownership model in C++ can be broadly categorized into two types: ownership of sub-objects stored in object fields and ownership of dynamically allocated objects using pointers.
+
+### Ownership of Sub-objects Stored in Object Fields
+
+In C++, classes often contain member variables that represent sub-objects. These sub-objects are typically owned by the parent object, and their memory is automatically managed by the parent's lifecycle. When a class instance is created, its member variables are initialized, and when the instance is destroyed, the destructors of its member variables are automatically called.
```

Comment:
Maybe it is worth mentioning here, that constructors and destructors of sub-objects are called in specific order.